### ISSUE TITLE
[Bug]: Wrong font color in email template

### ISSUE THREAD

**Issue by CMiksche** [zitadel/zitadel#4855]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [ ] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Describe the bug

I setup custom branding for the light and dark mode. Now I send a test verification mail to my own user address. The font color is white despite the primary color being also a bright color...

### To reproduce

1. Setup custom branding, choose a bright primary color
2. Send a verification mail
3. Zitadel will use the primary color in the mail as background of the button with a white text which makes it hard to read

### Screenshots

<img width="435" alt="Screenshot 2022-12-08 at 14 42 35" src="https://user-images.githubusercontent.com/9869391/206461335-b8c288ed-2b91-4eca-a8f7-a93f38d024c9.png">

### Expected behavior

If the email is in the light mode, it should also take the same colours like setup in the branding settings - which would be black text in my case.

### Version

2.14.3

### Operating System

Irrelevant

### Relevant Configuration

Light Mode Config:

<img width="287" alt="Screenshot 2022-12-08 at 14 44 23" src="https://user-images.githubusercontent.com/9869391/206461781-ed902645-8d38-4413-ac5a-360f40b4fa4f.png">

Dark Mode Config:
<img width="287" alt="Screenshot 2022-12-08 at 14 44 56" src="https://user-images.githubusercontent.com/9869391/206461852-fa262422-e63b-4831-9b62-e4ece92c62bd.png">

**Comment by CMiksche** [comment]
Is connected with #4191

**Comment by stebenz** [comment]
@hifabienne 
The email template part of the button contains `color:#ffffff`, which is white,  not quite sure if this should be changed and to what? Make it configurable or change it to the same color as the text?

**Comment by stebenz** [comment]
@hifabienne in the old and new login, the color of the text in the buttons gets calculated through contrast to the button color, so this should be done the same?
As an example: https://github.com/zitadel/typescript/blob/main/apps/login/src/helpers/colors.ts#L175

**Comment by hifabienne** [comment]
@skewis6 What is your opinion on this.

My thought:
1. it should work the same everywhere
2. We should take the colors the user can define, and not generate something else

So not sure what would be the best approach here.

**Comment by skewis6** [comment]
> [@skewis6](https://github.com/skewis6) What is your opinion on this.
> 
> My thought:
> 
> 1. it should work the same everywhere
> 2. We should take the colors the user can define, and not generate something else
> 
> So not sure what would be the best approach here.

Hmm yeah tricky one. I would probably do both @hifabienne 

- if the user has not provided a font colour
   - use the contrast @stebenz mentioned
- if the user has provided a font colour
   - use the font colour the user has provided

Readability and contrast is important, so I would like to include that everywhere. BUT users should be able to override that by providing a colour. 

We should probably enhance the UI to inform users of readability issues with their selected colours - if there are any.

**Comment by hifabienne** [comment]
> > [@skewis6](https://github.com/skewis6) What is your opinion on this.
> > My thought:
> > 
> > 1. it should work the same everywhere
> > 2. We should take the colors the user can define, and not generate something else
> > 
> > So not sure what would be the best approach here.
> 
> Hmm yeah tricky one. I would probably do both [@hifabienne](https://github.com/hifabienne)
> 
> * if the user has not provided a font colour
>   
>   * use the contrast [@stebenz](https://github.com/stebenz) mentioned
> * if the user has provided a font colour
>   
>   * use the font colour the user has provided
> 
> Readability and contrast is important, so I would like to include that everywhere. BUT users should be able to override that by providing a colour.
> 
> We should probably enhance the UI to inform users of readability issues with their selected colours - if there are any.

Should the font color also trigger for the primary buttons? Looking at the above dark theme, this could result in some problems depending on the colors you pick.

**Comment by skewis6** [comment]
> > > [@skewis6](https://github.com/skewis6) What is your opinion on this.
> > > My thought:
> > > 
> > > 1. it should work the same everywhere
> > > 2. We should take the colors the user can define, and not generate something else
> > > 
> > > So not sure what would be the best approach here.
> > 
> > Hmm yeah tricky one. I would probably do both [@hifabienne](https://github.com/hifabienne)
> > 
> > * if the user has not provided a font colour
> >   
> >   * use the contrast [@stebenz](https://github.com/stebenz) mentioned
> > * if the user has provided a font colour
> >   
> >   * use the font colour the user has provided
> > 
> > Readability and contrast is important, so I would like to include that everywhere. BUT users should be able to override that by providing a colour.
> > 
> > We should probably enhance the UI to inform users of readability issues with their selected colours - if there are any.
> 
> Should the font color also trigger for the primary buttons? Looking at the above dark theme, this could result in some problems depending on the colors you pick. 

No, we should have another field for button font colour in my opinion. It doesn't make sense to have them the same.

**Comment by elinashoko** [comment]
Thanks for filing this issue. As part of an ongoing backlog cleanup, we're closing bug reports and requests filed against ZITADEL v2.x, since that's several major versions behind current (v4.x, with v5 in progress).

If you're still seeing this on a current version, let us know in a comment with your updated version info (or reopen the issue) and we'll take another look.

This issue will close in 30 days if there's no further activity.

**Comment by elinashoko** [comment]
Thanks for filing this issue. As part of an ongoing backlog cleanup, we're closing issues that have had no activity in over 12 months.

If this is still relevant, let us know in a comment (or reopen the issue) and we'll take another look.

This issue will close in 30 days if there's no further activity.
